File: src/array/lc26.java
package array;

public class lc26 {
	public static void main(String[] args) {
		System.out.println(removeDuplicates(new int[] { 1, 2, 2, 3, 3, 4 }));
	}

	public static int removeDuplicates(int[] nums) {
		// 使用双指针
		if (nums == null || nums.length == 1) {
			return nums.length;
		}
		int i = 0, j = 1;
		while (j < nums.length) {
			if (nums[i] == nums[j]) {
				j++;
			} else {
				i++;
				nums[i] = nums[j];
				j++;
			}
		}
		return i + 1;
	}
}

File: c++/array/lc26.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int len = nums.size();
        if(len==0||len==1)	return len;
		int now = 0;
        for(int i=1;i<n;i++){
        	if(nums[i]==nums[now]){
        		continue;
			}else{
				now++;
				nums[i]=nums[now];
			}
		}
		return now+1;
    }
};
